### PR TITLE
Add unit tests for various PSAppDeployToolkit functions

### PR DESCRIPTION
**Pull request by sintaxasn**
# Pull Request

## Description

Added unit tests for the following functions:
- Add-ADTModuleCallback - tests callback addition, duplicate prevention, ordering and read-only results.
- Clear-ADTModuleCallback - tests clearing callbacks, empty hookpoints, independence, and reuse after clear.
- Complete-ADTFunction - tests successful completion behavior and logging at function end.
- Convert-ADTValueType - tests numeric type conversions, wrapping behavior, aliases, return types, pipeline input and validation.
- Convert-ADTValuesFromRemainingArguments - tests parsing remaining arguments into a dictionary.
- ConvertTo-ADTNTAccountOrSID - tests NTAccount/SID conversions, well-known SIDs and round-trips.
- Copy-ADTContentToCache - tests cache folder creation, copy behavior, session updates, and error handling.
- Disable-ADTTerminalServerInstallMode - tests safe behavior when disabling install mode.
- Enable-ADTTerminalServerInstallMode - tests safe behavior when enabling install mode and early-return cases.
- Export-ADTEnvironmentTableToSessionState - tests exporting environment table values into session state.
- Get-ADTApplication - tests installed application queries, filtering, matching modes, and update inclusion.
- Get-ADTBoundParametersAndDefaultValues - tests collecting bound parameters plus default values from invocation metadata.
- Get-ADTCommandTable - tests command table contents, return type and read-only enforcement.
- Get-ADTConfig - tests config access in initialized and uninitialized states.
- Get-ADTDeferHistory - tests defer-history retrieval and error propagation.
- Get-ADTEnvironmentTable - tests environment table access and seeded values.
- Get-ADTEnvironmentVariable - tests environment variable lookup, targets and validation.
- Get-ADTExecutableInfo - tests executable metadata lookup across parameter sets and architectures.
- Get-ADTFileVersion - tests file/product version extraction and invalid file handling.
- Get-ADTFreeDiskSpace - tests disk space retrieval and return formatting.
- Get-ADTLoggedOnUser - tests logged-on session enumeration and returned session properties.
- Get-ADTModuleCallback - tests callback retrieval, types, counts and hookpoint isolation.
- Get-ADTMsiExitCodeMessage - tests MSI exit code message lookup for known and missing messages.
- Get-ADTMsiTableProperty - tests MSI property and summary info reading.
- Get-ADTObjectProperty - tests reflective property access, indexed properties, and error cases.
- Get-ADTOperatingSystemInfo - tests OS info return type, key properties, and singleton behavior.
- Get-ADTPEFileArchitecture - tests PE architecture detection.
- Get-ADTPendingReboot - tests reboot info object shape and property types.
- Get-ADTPowerShellProcessPath - tests PowerShell executable path resolution and determinism.
- Get-ADTPresentationSettingsEnabledUsers - tests retrieval of users with presentation settings enabled.
- Get-ADTRunningProcesses - tests matching running processes from process definitions.
- Get-ADTStringTable - tests string table access in initialized and uninitialized states.
- Get-ADTUserNotificationState - tests bypass behavior when no active user is present.
- Get-ADTUserProfiles - tests user profile enumeration, filtering, exclusion and SID lookup.
- Get-ADTWindowTitle - tests no-user bypass path and input validation.
- Invoke-ADTAllUsersRegistryAction - tests all-user registry action execution and $_ binding.
- Invoke-ADTCommandWithRetries - tests retries, argument forwarding, and time-related validation.
- Invoke-ADTFunctionErrorHandler - tests error propagation, logging, and activity rewriting.
- Invoke-ADTObjectMethod - tests reflective method invocation with positional and named arguments.
- Invoke-ADTRegSvr32 - tests validation and around DLL registration and unregistration.
- New-ADTErrorRecord - tests building error records with required and optional metadata.
- New-ADTLogFileName - tests log filename generation with and without active session context.
- New-ADTValidateScriptErrorRecord - tests validate-script error record construction and formatting.
- New-ADTZipFile - tests zip creation, overwrite behavior.
- Out-ADTPowerShellEncodedCommand - tests UTF-16 / Base64 PowerShell encoded-command generation.
- Register-ADTDll - tests DLL registration validation.
- Remove-ADTContentFromCache - tests cache removal, session resets.
- Remove-ADTDesktopShortcut - tests desktop shortcut deletion by mode, filter and scope.
- Remove-ADTEdgeExtension - tests removal behavior when Edge extension config is present or absent.
- Remove-ADTEnvironmentVariable - tests process-scope environment variable removal.
- Remove-ADTFile - tests file/folder deletion paths, wildcards, recursion and arrays.
- Remove-ADTFileFromUserProfiles - tests removing paths across user profiles, including wildcard and recurse modes.
- Remove-ADTFolder - tests folder deletion, wildcard support, non-recursive behavior.
- Remove-ADTInvalidFileNameChars - tests invalid filename character stripping and whitespace trimming.
- Remove-ADTModuleCallback - tests removing callbacks singly and in batches.
- Remove-ADTRegistryKey - tests registry key/value deletion and recursion.
- Reset-ADTDeferHistory - tests defer-history reset delegation and error propagation.
- Resolve-ADTErrorRecord - tests formatting resolved error output and option switches.
- Set-ADTDeferHistory - tests defer-history updates and argument forwarding to session methods.
- Set-ADTEnvironmentVariable - tests process-scope environment variable creation / update.
- Set-ADTItemPermission - tests ACL updates, inheritance control, and direct ACL application.
- Set-ADTPowerShellCulture - tests culture validation against installed languages.
- Start-ADTProcess - tests process launch behavior, pass-through, exit codes, working directory.
- Test-ADTBattery - tests battery/power-state boolean and pass-through detail object.
- Test-ADTEspActive - tests ESP detection return type, logging and common machine states.
- Test-ADTMSUpdates - tests KB detection via mocked hotfix lookup.
- Test-ADTMicrophoneInUse - tests microphone-in-use detection and logging.
- Test-ADTModuleInitialized - tests module initialized-state reporting.
- Test-ADTNetworkConnection - tests active network detection by adapter type.
- Test-ADTOobeCompleted - tests OOBE completion reporting.
- Test-ADTPowerPoint - tests PowerPoint presentation-state detection across several scenarios.
- Test-ADTSessionActive - tests active-session detection.
- Test-ADTUserIsBusy - tests user-busy detection from multiple delegate checks.
- Unregister-ADTDll - tests DLL unregistration validation and delegation.
- Update-ADTDesktop - tests desktop refresh behavior with and without active user context.
- Update-ADTEnvironmentPsProvider - tests environment PS provider refresh behavior.
- Update-ADTGroupPolicy - tests group policy update execution paths and target handling.

Provided as part of #1414

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] General code cleanup (non-breaking change which improves readability)

## Checklist

- [X] I am pulling to the **main** branch
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have tested my changes to prove my fix is effective
- [X] I have tested that the module can build following my changes
- [X] I have made sure that any script file-encoding is set to UTF8 with BOM, i.e. unchanged.

## How Has This Been Tested?

Included as part of the unit testing, and verified that build succeeds.

### CODE REVIEW

File: src/PSAppDeployToolkit.Build/Tests/Unit/Enable-ADTTerminalServerInstallMode.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    Context 'On a Terminal Server already in install mode' {
+        It 'Does not throw when already in install mode (returns early)' {
+            $inInstallMode = [PSADT.TerminalServices.TerminalServerUtilities]::InAppInstallMode()
+            if (!$inInstallMode)
+            {
+                Set-ItResult -Skipped -Because 'This system is not currently in Terminal Server install mode'
+                return
+            }
+            # InAppInstallMode() = $true → "already in install mode" → returns early without throw.
+            { Enable-ADTTerminalServerInstallMode } | Should -Not -Throw
```
Comment (michael-hollingsworth) [inline src/PSAppDeployToolkit.Build/Tests/Unit/Enable-ADTTerminalServerInstallMode.Tests.ps1]: Same as the comment in `Disable-ADTTerminalServerInstallMode`.

File: src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTApplication.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        }
+
+        It 'Returns at least one installed application' {
+            $result = @(Get-ADTApplication)
+            $result.Count | Should -BeGreaterThan 0
+        }
+
+        It 'Each result is of type PSADT.AppManagement.InstalledApplication' {
+            $result = @(Get-ADTApplication)
+            $result[0] | Should -BeOfType ([PSADT.AppManagement.InstalledApplication])
```
Comment (michael-hollingsworth) [inline src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTApplication.Tests.ps1]: This doesn't check each result like the test implies, just the first one.

File: src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTApplication.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+        It 'Returns empty or fewer results for a nonexistent app name' {
+            $result = @(Get-ADTApplication -Name 'PSADT_NonExistent_App_ZZZ')
+            $result.Count | Should -Be 0
+        }
+    }
+
+    Context '-NameMatch Exact' {
+        It 'Does not throw with -NameMatch Exact and a name that does not match exactly' {
+            { Get-ADTApplication -Name 'Microsoft' -NameMatch Exact } | Should -Not -Throw
```
Comment (michael-hollingsworth) [inline src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTApplication.Tests.ps1]: Realistically, this is a valid test but, I'd like to see some logic to validate that an application with that name doesn't exist before calling `Get-ADTApplication`. You can continue with the name `Microsoft` or, you could take a real application name from the system and remove one or more characters from the DisplayName.

File: src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTBoundParametersAndDefaultValues.Tests.ps1 [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        Mock -ModuleName PSAppDeployToolkit Set-ADTPreferenceVariables { }
+        # Mock Write-ADTLogEntry due to its expense when running via Pester.
+        Mock -ModuleName PSAppDeployToolkit Write-ADTLogEntry {}
+
+        # Test helpers: real named functions so that $MyInvocation.MyCommand.ScriptBlock.Ast
+        # is a FunctionDefinitionAst that Get-ADTBoundParametersAndDefaultValues can parse.
+        function script:Invoke-BoundParamHelper
+        {
+            [CmdletBinding()]
+            param(
```
Comment (michael-hollingsworth) [inline src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTBoundParametersAndDefaultValues.Tests.ps1]: Use Allman bracing

File: src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTBoundParametersAndDefaultValues.Tests.ps1 [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+            )
+            $null = $Name, $Count
+            return Get-ADTBoundParametersAndDefaultValues -Invocation $MyInvocation -Exclude 'Name'
+        }
+    }
+
+    Context 'Default Values' {
+        It 'Includes a parameter with a default string value when not explicitly bound' {
+            $result = Invoke-BoundParamHelper
+            $result.ContainsKey('Name') | Should -BeTrue
```
Comment (michael-hollingsworth) [inline src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTBoundParametersAndDefaultValues.Tests.ps1]: Would a custom `-ContainsKey` assertion be better here? And, throughout the rest of these tests.

File: src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTCommandTable.Tests.ps1 [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+            $result = Get-ADTCommandTable
+            $result | Should -BeOfType ([System.Collections.Generic.IReadOnlyDictionary[System.String, System.Management.Automation.CommandInfo]])
+        }
+
+        It 'The runtime type is a ReadOnlyDictionary wrapper' {
+            $result = Get-ADTCommandTable
+            $result.GetType().Name | Should -BeLike '*ReadOnly*'
+        }
+
+        It 'Does not throw when called' {
```
Comment (michael-hollingsworth) [inline src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTCommandTable.Tests.ps1]: I'm not sure that these generic `Does not throw when called` tests add anything to our test suite.

File: src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTEnvironmentVariable.Tests.ps1 [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+            { Get-ADTEnvironmentVariable -Variable 'SystemRoot' -Target Process } | Should -Not -Throw
+        }
+    }
+
+    Context 'Input Validation' {
+        It 'Throws when Variable is null' {
+            { Get-ADTEnvironmentVariable -Variable $null } | Should -Throw
+        }
+
+        It 'Throws when Variable is an empty string' {
```
Comment (michael-hollingsworth) [inline src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTEnvironmentVariable.Tests.ps1]: The other null/whitespace values used in some of our other tests should also be tested against.

File: src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTExecutableInfo.Tests.ps1 [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    Context 'Path parameter set' {
+        It 'Does not throw using -Path with a valid executable path' {
+            { Get-ADTExecutableInfo -Path $script:Cmd64 } | Should -Not -Throw
+        }
+
+        It 'Returns a non-null result using -Path' {
+            $result = Get-ADTExecutableInfo -Path $script:Cmd64
+            $result | Should -Not -BeNull
```
Comment (michael-hollingsworth) [inline src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTExecutableInfo.Tests.ps1]: While this works, `-BeNull` isn't documented at all.

File: src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTFreeDiskSpace.Tests.ps1 [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+        }
+    }
+
+    Context 'Explicit Drive' {
+        It 'Returns a non-negative value for the system drive specified explicitly' {
+            $result = Get-ADTFreeDiskSpace -Drive ([System.IO.DriveInfo]$env:SystemDrive)
+            $result | Should -BeGreaterOrEqual 0
+        }
+
+        It 'Does not throw when an explicit valid drive is supplied' {
```
Comment (michael-hollingsworth) [inline src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTFreeDiskSpace.Tests.ps1]: If we are to keep this test, it should be moved to the top of this context block.

File: src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTFreeDiskSpace.Tests.ps1 [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+        # Mock Set-ADTPreferenceVariables to avoid changing preference state during tests.
+        Mock -ModuleName PSAppDeployToolkit Set-ADTPreferenceVariables { }
+        # Mock Write-ADTLogEntry due to its expense when running via Pester.
+        Mock -ModuleName PSAppDeployToolkit Write-ADTLogEntry { }
+    }
+
+    Context 'Default Drive (System Drive)' {
+        It 'Returns a non-negative value when called without a Drive parameter' {
+            $result = Get-ADTFreeDiskSpace
+            $result | Should -BeGreaterOrEqual 0
```
Comment (michael-hollingsworth) [inline src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTFreeDiskSpace.Tests.ps1]: I've run into computers where the value returned is 0.

File: src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTFreeDiskSpace.Tests.ps1 [hunk 11]
```diff
@@ -0,0 +1,47 @@
+﻿BeforeAll {
+    Remove-Module PSAppDeployToolkit -Force -ErrorAction SilentlyContinue
+    Import-Module "$PSScriptRoot\..\..\..\PSAppDeployToolkit\PSAppDeployToolkit.psd1" -Force
+}
+
+Describe 'Get-ADTFreeDiskSpace' {
```
Comment (michael-hollingsworth) [inline src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTFreeDiskSpace.Tests.ps1]: This is missing a test to validate that the results returned by `Get-ADTFreeDiskSpace` accurately represent the remaining available space on a drive.

File: src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTModuleCallback.Tests.ps1 [hunk 12]
```diff
@@ -0,0 +1,96 @@
+﻿BeforeAll {
```
Comment (michael-hollingsworth) [inline src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTModuleCallback.Tests.ps1]: Not necessarily an issue but, since nothing is mocked and this test utilizes 3 of the 4 `*-ADTModuleCallback` functions, it feels like more of an integration test than a unit test. Same applies to the other `*-ADTModuleCallback` functions.

File: src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTMsiExitCodeMessage.Tests.ps1 [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+    Context 'Exit Codes with No Message in msimsg.dll' {
+        It 'Returns no output for exit code 1602 (no DLL resource on this platform)' {
+            # 1602 has no string resource in msimsg.dll on this machine — function returns nothing.
+            $result = Get-ADTMsiExitCodeMessage -MsiExitCode 1602
+            $result | Should -BeNullOrEmpty
+        }
+    }
+
+    Context 'Does Not Throw' {
+        It 'Does not throw for exit code 1618' {
```
Comment (michael-hollingsworth) [inline src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTMsiExitCodeMessage.Tests.ps1]: This needs to be refactored or removed. The very first test calls `Get-ADTMsiExitCodeMessage -MsiExitCode 1618`.

File: src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTPEFileArchitecture.Tests.ps1 [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+        Mock -ModuleName PSAppDeployToolkit Write-ADTLogEntry { }
+    }
+
+    Context 'Return Type' {
+        It 'Returns a PSADT IMAGE_FILE_MACHINE enum value' {
+            $result = Get-ADTPEFileArchitecture -LiteralPath $script:Exe64
+            $result | Should -BeOfType ([PSADT.Interop.IMAGE_FILE_MACHINE])
+        }
+
+        It 'Does not throw for a valid 64-bit executable' {
```
Comment (michael-hollingsworth) [inline src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTPEFileArchitecture.Tests.ps1]: If we were to keep this, it should be moved to the top of this `Context`.

File: src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTPresentationSettingsEnabledUsers.Tests.ps1 [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+
+        It 'Any returned objects are of type PSADT.Types.UserProfileInfo' {
+            $result = @(Get-ADTPresentationSettingsEnabledUsers)
+            foreach ($item in $result)
+            {
+                $item | Should -BeOfType ([PSADT.Types.UserProfileInfo])
+            }
+        }
+
+        It 'Any returned objects have a non-null ProfilePath' {
```
Comment (michael-hollingsworth) [inline src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTPresentationSettingsEnabledUsers.Tests.ps1]: This test doesn't do a whole lot. If it is to be kept, it should be moved above the test preceding this one.

File: src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTUserProfiles.Tests.ps1 [hunk 16]
```diff
@@ -0,0 +1,107 @@
...
+    }
+
+    Context '-SID parameter set' {
+        It 'Does not throw when specifying the current user SID' {
+            $currentSID = [System.Security.Principal.WindowsIdentity]::GetCurrent().User
+            { Get-ADTUserProfiles -SID $currentSID } | Should -Not -Throw
+        }
+
+        It 'Returns the current user profile when specifying the current user SID' {
+            $currentSID = [System.Security.Principal.WindowsIdentity]::GetCurrent().User
```
Comment (michael-hollingsworth) [inline src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTUserProfiles.Tests.ps1]: Not a massive issue, just something to note: If the current identity is `NT AUTHORITY\SYSTEM`, a `UserProfileInfo` won't be returned.

File: src/PSAppDeployToolkit.Build/Tests/Unit/Invoke-ADTCommandWithRetries.Tests.ps1 [hunk 17]
```diff
@@ -0,0 +1,123 @@
...
+    Context 'Parameter Forwarding' {
+        It 'Forwards remaining arguments to the target command' {
+            $result = Invoke-ADTCommandWithRetries -Command 'Invoke-ADTTestEchoValue' -SleepDuration ([System.TimeSpan]::FromMilliseconds(10)) -Value 'hello'
+            $result | Should -Be 'hello'
+        }
+    }
+
+    Context 'SleepDuration Validation' {
+        It 'Throws when SleepDuration is zero' {
+            { Invoke-ADTCommandWithRetries -Command 'Invoke-ADTTestAlwaysSucceed' -SleepDuration ([System.TimeSpan]::Zero) } | Should -Throw
```
Comment (michael-hollingsworth) [inline src/PSAppDeployToolkit.Build/Tests/Unit/Invoke-ADTCommandWithRetries.Tests.ps1]: Also validate against type conversion:
- `-SleepDuration 0`
- `-SleepDuration '0'`
- `-SleepDuration ([System.Int64]0)`
- `-SleepDuration ([System.Decimal]::Zero)`
- etc.

File: src/PSAppDeployToolkit.Build/Tests/Unit/New-ADTLogFileName.Tests.ps1 [hunk 18]
```diff
@@ -0,0 +1,70 @@
...
+            $result | Should -Be "$($script:MockLogPath)\Setup_tool.log"
+        }
+
+        It 'The Discriminator value appears in the returned filename' {
+            $result = New-ADTLogFileName -Discriminator 'MyApp' -FileNameOnly
+            $result | Should -BeLike 'MyApp*'
+        }
+
+        It 'Does not throw when a session is active' {
+            { New-ADTLogFileName -Discriminator 'Setup' -FileNameOnly } | Should -Not -Throw
```
Comment (michael-hollingsworth) [inline src/PSAppDeployToolkit.Build/Tests/Unit/New-ADTLogFileName.Tests.ps1]: If this test is gonna stay, it should be the first one in this `Context` block.

File: src/PSAppDeployToolkit.Build/Tests/Unit/New-ADTLogFileName.Tests.ps1 [hunk 19]
```diff
@@ -0,0 +1,70 @@
...
+        }
+    }
+
+    Context 'Without Active Session' {
+        BeforeAll {
+            Mock -ModuleName PSAppDeployToolkit Get-ADTSession { throw [System.Exception]::new('No active session') }
```
Comment (michael-hollingsworth) [inline src/PSAppDeployToolkit.Build/Tests/Unit/New-ADTLogFileName.Tests.ps1]: Why use this instead of the error that is naturally thrown in `Get-ADTSession`?

File: src/PSAppDeployToolkit.Build/Tests/Unit/New-ADTZipFile.Tests.ps1 [hunk 20]
```diff
@@ -0,0 +1,90 @@
...
+        Mock -ModuleName PSAppDeployToolkit Set-ADTPreferenceVariables { }
+        # Mock Write-ADTLogEntry due to its expense when running via Pester.
+        Mock -ModuleName PSAppDeployToolkit Write-ADTLogEntry { }
+    }
+
+    BeforeEach {
+        # Fresh source directory and unique destination path per test.
+        $script:SrcDir = Join-Path $TestDrive (New-Guid)
+        $null = New-Item -Path $script:SrcDir -ItemType Directory
+        Set-Content -Path (Join-Path $script:SrcDir 'data.txt')  -Value 'hello'
```
Comment (michael-hollingsworth) [inline src/PSAppDeployToolkit.Build/Tests/Unit/New-ADTZipFile.Tests.ps1]: Use name parameters instead of positional ones. This applies to the other `Join-Path` calls in this suite of tests.

File: src/PSAppDeployToolkit.Build/Tests/Unit/New-ADTZipFile.Tests.ps1 [hunk 21]
```diff
@@ -0,0 +1,90 @@
...
+        It 'Produces no pipeline output' {
+            $result = New-ADTZipFile -LiteralPath (Join-Path $script:SrcDir 'data.txt') -DestinationPath $script:ZipPath
+            $result | Should -BeNullOrEmpty
+        }
+
+        It 'Does not throw for a valid source file and destination' {
+            { New-ADTZipFile -LiteralPath (Join-Path $script:SrcDir 'data.txt') -DestinationPath $script:ZipPath } | Should -Not -Throw
+        }
+
+        It 'Source file still exists when -RemoveSourceAfterArchiving is not specified' {
```
Comment (michael-hollingsworth) [inline src/PSAppDeployToolkit.Build/Tests/Unit/New-ADTZipFile.Tests.ps1]: What about a test that validates the file is removed when `-RemoveSourceAfterArchiving` is specified?

File: src/PSAppDeployToolkit.Build/Tests/Unit/Out-ADTPowerShellEncodedCommand.Tests.ps1 [hunk 22]
```diff
@@ -0,0 +1,92 @@
...
+        }
+
+        It 'Returns a valid Base64 string (no throw when decoding)' {
+            $encoded = Out-ADTPowerShellEncodedCommand -Command 'Get-Process'
+            { [System.Convert]::FromBase64String($encoded) } | Should -Not -Throw
+        }
+    }
+
+    Context 'Unicode Encoding' {
+        It 'Uses UTF-16LE (Unicode) encoding — result length is a multiple of 4' {
```
Comment (michael-hollingsworth) [inline src/PSAppDeployToolkit.Build/Tests/Unit/Out-ADTPowerShellEncodedCommand.Tests.ps1]: I'm not sure that this is a feature of this function, rather an unintended side affect.

**Comment by sintaxasn** [comment]
Closing this PR as I have an alternate currently in draft.

File: src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTPresentationSettingsEnabledUsers.Tests.ps1 [hunk 23]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+Describe 'Get-ADTPresentationSettingsEnabledUsers' {
+    BeforeAll {
+        # Mock Set-ADTPreferenceVariables to avoid changing preference state during tests.
+        Mock -ModuleName PSAppDeployToolkit Set-ADTPreferenceVariables { }
+        # Mock Write-ADTLogEntry due to its expense when running via Pester.
+        Mock -ModuleName PSAppDeployToolkit Write-ADTLogEntry { }
+    }
+
```
[the remark this answers is not in the document]
Comment (DanGough) [reply src/PSAppDeployToolkit.Build/Tests/Unit/Get-ADTPresentationSettingsEnabledUsers.Tests.ps1:13]: Dismissing, in fact I am removing the IsAdmin checks from other tests as they are not required since the functions work without elevation.
